File: 백준/Silver/10828. 스택/스택.py
import sys
tc = int(sys.stdin.readline().rstrip())
stack = []

for _ in range(tc):
    cmd = sys.stdin.readline().rstrip() # 명령어 입력받기
    if 'push' in cmd:
        p, num = cmd.split()
        stack.append(int(num))
    elif cmd == 'pop':
        if not stack:
            print(-1)
        else:
            print(stack.pop())
    elif cmd == 'size':
        print(len(stack))
    elif cmd == 'empty':
        if not stack: #비어있으면
            print(1)
        else:
            print(0)
    elif cmd == 'top':
        if not stack:
            print(-1)
        else:
            print(stack[-1])
    else:
        print("invalid command")

File: 백준/Silver/10828. 스택/스택.js
const input = require("fs")
    .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./input.txt")
    .toString()
    .trim()
    .split("\n")
    .map((v) => v.split(" "));
const N = +input[0][0];
const stack = [];
const answer = [];

for (let i = 0; i < N; i++) {
    const [command] = input[i+1];
    switch (command) {
        case 'push':
            let num = +input[i+1][1];
            stack.push(num);
            break;
        case 'pop':
            if (stack.length === 0) {
                answer.push(-1);
                break;
            } else {
                const num = stack.pop();
                answer.push(num);
                break;
            }
        case 'size':
            answer.push(stack.length);
            break;
        case 'empty':
            answer.push(stack.length === 0 ? 1 : 0);
            break;
        case 'top':
            answer.push(stack.length ===  0 ? -1 : stack.at(-1));
            break;
        default:
            break;
    }
}

console.log(answer.join('\n'));
